my-app/src/app/core/interceptors/loading.interceptor.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpResponse,
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+  HttpInterceptor,
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { LoadingService } from '../service/loading.service';
+
+@Injectable()
+export class LoaderInterceptor implements HttpInterceptor {
+  private requests: HttpRequest<any>[] = [];
+
+  constructor(private LoadingService: LoadingService) {}
+
+  public removeRequest(req: HttpRequest<any>) {
+    const i = this.requests.indexOf(req);
+    if (i >= 0) {
+      this.requests.splice(i, 1);
+    }
+    this.LoadingService.isLoading.next(this.requests.length > 0);
+  }
+
+  public intercept(
+    req: HttpRequest<any>,
+    next: HttpHandler
+  ): Observable<HttpEvent<any>> {
+    this.requests.push(req);
```

Comment:
why do you accumulate requests? just toggle the pinned and hide it when the response comes.

Do not increase complexity. 
There are lots of articles how to implement it.
https://medium.com/swlh/angular-loading-spinner-using-http-interceptor-63c1bb76517b

here is the one. Have a look at HttpRequestInterceptor and how to toggles the loading
You can ignore loadingMap and just show and hide the spinner. You won't have the parallel request anyway